File: scripts/read-pdf.py
import PyPDF2
import sys

def read_pdf(file_path):
    try:
        with open(file_path, 'rb') as file:
            reader = PyPDF2.PdfReader(file)
            text = ""
            for page_num in range(len(reader.pages)):
                text += reader.pages[page_num].extract_text()
            print("----- PDF TEXT START -----")
            print(text)
            print("----- PDF TEXT END -----")
    except Exception as e:
        print(f"Error reading PDF: {e}")

if __name__ == "__main__":
    if len(sys.argv) > 1:
        read_pdf(sys.argv[1])
    else:
        read_pdf('public/documents/Aflah_Muhammed.pdf')

File: scripts/read-pdf.ts
import fs from 'fs';
import pdf from 'pdf-parse';

async function readPdf() {
    try {
        let dataBuffer = fs.readFileSync('public/documents/Aflah_Muhammed.pdf');
        const data = await pdf(dataBuffer);
        console.log("----- PDF TEXT START -----");
        console.log(data.text);
        console.log("----- PDF TEXT END -----");
    } catch (e) {
        console.error("Error parsing", e);
    }
}
readPdf();
